Dev databoxedge microsoft.data box edge 2020 07 01 preview

pvsraviteja opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

pvsraviteja committed:
Adds base for updating Microsoft.DataBoxEdge from version preview/2020-05-01-preview to version 2020-07-01-preview

pvsraviteja committed:
Updates readme

pvsraviteja committed:
Updates API version in new specs and examples

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?

pvsraviteja committed:
Update databoxedge.json

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Example file for GetDCAccessCode.json

GetDCAccessCode example file

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Change to camel case

Change to camel case

pvsraviteja committed:
Fix prettier validation

Fix prettier validation

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Update databoxedge.json

Update databoxedge.json to eliminate validation failures

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Fix violation in examples file

Fix violation in examples file

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Update example to fix Model violation

Update example to fix Model violation

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Revert json files to pass Prettier validation

Revert json files to pass Prettier validation

pvsraviteja committed:
Changes to DCAccessCode

Changes to DCAccessCode

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Update DataBoxEdge json

Update DataBoxEdge json

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Revert "Update DataBoxEdge json"

This reverts commit b86243eedaa770960c015b284f06de299177d0d4.

pvsraviteja committed:
Update DataBoxEdge json

Update DataBoxEdge json

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Change json files

Change json files

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Update GetDCAccessCode.json

Update GetDCAccessCode.json

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Fix Model violation

Fix Model violation

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Update properties in DataBoxEdge.json

AutorestCI commented:
/azp run automation - sdk

weidongxu-microsoft commented:
API change:
https://github.com/Azure/azure-rest-api-specs/pull/10044/files/8de2e9debff7b404dd711b70b1b8159f98e0c2e2..b73c6ee8b1b65590bb331cedba4286411584b6d8

weidongxu-microsoft commented on specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:
```diff
@@ -136,6 +136,52 @@
+          {
+            "name": "$expand",
+            "in": "query",
+            "description": "Specify $filter='location eq <location>' to filter on location resource or Specify $skipToken=<token> to populate the next page in the list.",
```
It is `$expand`, why description is about filter and skipToken?

pvsraviteja replied:
Changed it.

weidongxu-microsoft commented on specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:
```diff
@@ -136,6 +136,52 @@
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
```
`DataBoxEdgeSkuList` has `nextLink` property. Is it correct here as `null`?

pvsraviteja replied:
This API is not used. But changed it.

weidongxu-microsoft commented on specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:
```diff
@@ -3920,6 +3979,31 @@
           "type": "integer",
           "readOnly": true
+        },
+        "resourceMoveDetails": {
```
Is it `readonly`?

pvsraviteja replied:
Yes changed it.

pvsraviteja committed:
Update DataBoxEdge.json

Update DataBoxEdge.json with the comments

AutorestCI commented:
/azp run automation - sdk

majastrz commented:
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/getDCAccessCode": {

if you call it listDCAccessCode, it will integrate nicely with ARM templates.

---
Refers to: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:951 in 7546fc3. [](commit_id = 7546fc3a4534b0679eaa308bef1d29faf380d78b, deletion_comment = False)

majastrz reviewed (approved):
Signed off from ARM side with a comment. If getting the access code doesn't make sense in a template, it's fine as is.

weidongxu-microsoft commented:
@pvsraviteja

Please check ARM's feedback and see if you need to act on it.

When you are all done, let me know when you want the PR merged.

pvsraviteja committed:
Change Tag Name for DCAccessCode API

Change Tag Name for DCAccessCode API

AutorestCI commented:
/azp run automation - sdk

pvsraviteja commented:
> Signed off from ARM side with a comment. If getting the access code doesn't make sense in a template, it's fine as is.

Getting accesscode is an action and it doesn't fit with ARM template I think

AutorestCI commented:
/azp run automation - sdk

pvsraviteja commented:
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/getDCAccessCode": {
> > ```
> 
> if you call it listDCAccessCode, it will integrate nicely with ARM templates.
> 
> Refers to: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:951 in [7546fc3](https://github.com/Azure/azure-rest-api-specs/commit/7546fc3a4534b0679eaa308bef1d29faf380d78b). [](commit_id = [7546fc3](https://github.com/Azure/azure-rest-api-specs/commit/7546fc3a4534b0679eaa308bef1d29faf380d78b), deletion_comment = False)

I think GetDCAccessCode doesn't fit with ARM Template.

majastrz commented:
> I think GetDCAccessCode doesn't fit with ARM Template.

ARM templates allow using the `listKeys()` function to access the keys of a storage accounts - it is a very common scenario in ARM templates. (Behind the scenes, it translates to a POST request to the RP.) This is somewhat documented at https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list. 

Renaming an action in the future will be a breaking change which requires separate approvals these days. As such, it's better to get those changes in as early as possible to avoid it. Leaving it up to you, though.

pvsraviteja committed:
Change from Devices to orders for SDK

Change from Devices to orders for SDK

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Mistake in the Order Path

Mistake in the Order Path

pvsraviteja committed:
Change to ListDCAccessCode from GetDCAccessCode

Change to ListDCAccessCode from GetDCAccessCode to make it work through ARM templates.

AutorestCI commented:
/azp run automation - sdk

pvsraviteja committed:
Change fields to optional

Change fields to optional

AutorestCI commented:
/azp run automation - sdk

weidongxu-microsoft commented:
@pvsraviteja
Let me know when you plan to merge the PR.

pvsraviteja committed:
Update databoxedge.json

AutorestCI commented:
/azp run automation - sdk

weidongxu-microsoft commented:
@pvsraviteja
Let me know when you plan to merge the PR.

Also there is now conflict in readme.md

pvsraviteja commented:
Closing this as it's not relevant.
